ui/src/pages/namespace/Events/History/EventsList.tsx

```diff
@@ -109,29 +126,56 @@ const EventsList = ({
                   )}
                 </EventsTable>
               </Card>
-              {totalPages > 1 && (
-                <Pagination>
-                  <PaginationLink
-                    data-testid="pagination-btn-left"
-                    icon="left"
-                    onClick={() => goToPreviousPage()}
-                  />
-                  {pagesList.map((page) => (
-                    <PaginationLink
-                      active={currentPage === page}
-                      key={`${page}`}
-                      onClick={() => goToPage(page)}
+              <div className="flex items-center justify-end">
+                <div className="m-2">
+                  <Select
```

Comment:
I think we could make this Select its own component to make this file a bit cleaner. Just move it to `src/pages/namespace/Events/History/components/`. It would only need the `goToPage` callback as a prop.